Tests/test_health.gd

```diff
@@ -149,3 +153,64 @@ func test_poison_status():
 
 	poison_status.on_turn_start()
 	assert_eq(_enemy_health_component.current_health, 99.0)
+	
+func test_card_damage_all():
+	_enemy.get_party_component().set_party([_enemy, _enemy_2])
+	var card_damage_all: CardBase = load("res://Cards/Resource/Card_DamageAll.tres")
+	
+	card_damage_all.on_card_play(_player, _enemy)
+	assert_eq(_enemy_health_component.current_health, 98.0)
+	assert_eq(_enemy_2.get_health_component().current_health, 98.0)
+
+func test_card_damage():
+	var card_damage: CardBase = load("res://Cards/Resource/Card_Damage.tres")
+	card_damage.on_card_play(_player, _enemy)
+	
+	assert_eq(_enemy_health_component.current_health, 97.0)
+
+func test_card_damage_health():
+	_player.get_health_component()._set_health(90.0)
+	var card_damage_health: CardBase = load("res://Cards/Resource/Card_DamageHealth.tres")
+	card_damage_health.on_card_play(_player, _enemy)
+	
+	assert_eq(_enemy_health_component.current_health, 90.0)
+
+func test_card_poison():
+	var card_poison: CardBase = load("res://Cards/Resource/Card_Poison.tres")
+	
+	assert_eq(_enemy.get_status_component().current_status.size(), 0)
+	card_poison.on_card_play(_player, _enemy)
+	assert_eq(_enemy.get_status_component().current_status.size(), 1)
+	
+	var status = _enemy.get_status_component().current_status[0]
+	assert_is(status, Debuff_Poison)
+	assert_eq(status.status_turn_duration, 3)
+	
+	_enemy.get_status_component().apply_turn_start_status()
+	# May need to update once we have a better direction of what to do for poison, currently
+	# it deals only 1 damage per turn
+	assert_eq(_enemy_health_component.current_health, 99.0)
+
+func test_card_damage_and_poison():
+	var card_damage_and_poison: CardBase = load("res://Cards/Resource/Card_damage_and_poison.tres")
```

Comment:
maybe adding the description of the card in the test would be good (so we don't need to go check the card)